### PR TITLE
Solve multiple deprecations

### PR DESCRIPTION
**Pull request by VincentLanglet**

### CODE REVIEW

File: src/Admin/PageAdmin.php [hunk 1]
```diff
@@ -167,6 +154,19 @@ public function setCacheManager(CacheManagerInterface $cacheManager)
         $this->cacheManager = $cacheManager;
     }
 
+    protected function configureBatchActions($actions): array
+    {
+        $actions = parent::configureBatchActions($actions);
+
+        $actions['snapshot'] = [
+            'label' => 'create_snapshot',
+            'translation_domain' => $this->getTranslationDomain(),
```
Comment (Hanmac) [inline src/Admin/PageAdmin.php]: isn't that unnecessary? the translation domain defaults to the Admins Translation domain anyway

File: src/Admin/PageAdmin.php [hunk 2]
```diff
@@ -266,8 +266,8 @@ protected function configureDatagridFilters(DatagridMapper $filter)
                 'field_options' => [
                     'required' => false,
                     'choices' => [
-                        'hybrid' => $this->trans('hybrid'),
-                        'cms' => $this->trans('cms'),
+                        'hybrid' => $this->getTranslator()->trans('hybrid', [], 'SonataPageBundle'),
```
Comment (Hanmac) [inline src/Admin/PageAdmin.php]: wouldn't it just be enough to remove the 'choice_translation_domain' part?

then the translator would translate it anyway?

File: src/Admin/SnapshotAdmin.php [hunk 3]
```diff
@@ -68,7 +68,7 @@ public function getBatchActions()
         $actions = parent::getBatchActions();
 
         $actions['toggle_enabled'] = [
-            'label' => $this->trans('toggle_enabled'),
+            'label' => $this->getTranslator()->trans('toggle_enabled', [], 'SonataPageBundle'),
```
Comment (Hanmac) [inline src/Admin/SnapshotAdmin.php]: wouldn't this be automatic translated by the translator anyway?

File: src/Admin/PageAdmin.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Admin/PageAdmin.php]: I just changed getBatchActions to configureBatchActions without changing the code inside, but I can do this indeed.

File: src/Admin/PageAdmin.php (hunk 2 above)
Comment (VincentLanglet) [reply src/Admin/PageAdmin.php]: Indeed, should be better knwo

File: src/Admin/SnapshotAdmin.php (hunk 3 above)
Comment (VincentLanglet) [reply src/Admin/SnapshotAdmin.php]: Indeed

**Review by jordisala1991 — APPROVED** [review]

File: src/Admin/PageAdmin.php [hunk 4]
```diff
@@ -266,10 +265,10 @@ protected function configureDatagridFilters(DatagridMapper $filter)
                 'field_options' => [
                     'required' => false,
                     'choices' => [
-                        'hybrid' => $this->trans('hybrid'),
-                        'cms' => $this->trans('cms'),
+                        'hybrid' => 'hybrid',
+                        'cms' => 'cms',
                     ],
-                    'choice_translation_domain' => false,
+                    'choice_translation_domain' => 'SonataPageBundle',
```
Comment (Hanmac) [inline src/Admin/PageAdmin.php:271]: i would have just removed that line and have it get the translation domain from the admin anyway

Comment (VincentLanglet) [reply src/Admin/PageAdmin.php:271]: To me, the default value is not the admin translation domain here, but the one from ChoiceType, which is message.

So it's safer this way

**Comment by Hanmac** [comment]
@VincentLanglet i think you missed this part:

https://github.com/sonata-project/SonataPageBundle/blob/c9c1fdce6e4025597c32616f72cc5b555d5ed0d0/src/Admin/BaseBlockAdmin.php#L182-L191

or was that on purpose?

**Comment by VincentLanglet** [comment]
Missed it. Do you want to do the pr ?

**Comment by Hanmac** [comment]
i will do later if you aren't faster

**Comment by Hanmac** [comment]
can this change be done in 3.x or should that be in 4.x? The function was already marked as ``@final``

**Comment by VincentLanglet** [comment]
> can this change be done in 3.x or should that be in 4.x? The function was already marked as `@final`

This can be done on 3.x because if you implements configurePersistentParameter, the `getPersistentParameter` method will give the same result than before.

**Pull request merged**